src/tools/gophish_export.py

```diff
@@ -253,6 +253,86 @@ def get_application(rawEvent):
+    campaign_ids = get_campaign_ids(api, assessment_id)
+    campaign_data_template = "campaign_data.json"
+    campaign_summary_json = assessment_id + "_campaign_data.json"
+    campaign_summary_textfile = (
+        assessment_id + "_summary_" + str(datetime.now()) + ".txt"
+    )
+
+    with open(campaign_data_template) as template:
+        campaign_data = json.load(template)
+
+    logging.info("Writing campaign summary report to %s" % campaign_summary_textfile)
+    file_out = open(campaign_summary_textfile, "w+")
+    file_out.write("Campaigns for Assessment: " + assessment_id)
+
+    for campaign_id in campaign_ids:
+        campaign = api.campaigns.get(campaign_id)
+        if campaign.name.endswith("_level-1"):
+            level = "level-1"
+        elif campaign.name.endswith("_level-2"):
+            level = "level-2"
+        elif campaign.name.endswith("_level-3"):
+            level = "level-3"
+        elif campaign.name.endswith("_level-4"):
+            level = "level-4"
+        elif campaign.name.endswith("_level-5"):
+            level = "level-5"
+        elif campaign.name.endswith("_level-6"):
+            level = "level-6"
+        else:
+            continue
+
+        logging.info(level)
+        clicks = get_click_data(api, campaign_id)
+
+        total_clicks = api.campaigns.summary(campaign_id=campaign_id).stats.clicked
+        unique_clicks = find_unique_target_clicks_count(clicks)
+        if total_clicks > 0:
+            percent_clicks = unique_clicks / float(total_clicks)
+        else:
+            percent_clicks = 0
+        campaign_data[level]["subject"] = campaign.template.subject
+        campaign_data[level]["sender"] = campaign.smtp.from_address
+        campaign_data[level]["start_date"] = campaign.launch_date
+        campaign_data[level]["end_date"] = campaign.completed_date
+        campaign_data[level]["redirect"] = campaign.url
+        campaign_data[level]["clicks"] = total_clicks
+        campaign_data[level]["unique_clicks"] = unique_clicks
+        campaign_data[level]["percent_clicks"] = percent_clicks
+
+        file_out.write("\n")
+        file_out.write("-" * 50)
+        file_out.write("\nCampaign: %s" % campaign.name)
+        file_out.write("\nSubject: %s" % campaign_data[level]["subject"])
+        file_out.write("\nSender: %s" % campaign_data[level]["sender"])
+        file_out.write("\nStart Date: %s" % campaign_data[level]["start_date"])
+        file_out.write("\nEnd Date: %s" % campaign_data[level]["end_date"])
+        file_out.write("\nRedirect: %s" % campaign_data[level]["redirect"])
+        file_out.write("\nClicks: %i" % campaign_data[level]["clicks"])
+        file_out.write("\nUnique Clicks: %i" % campaign_data[level]["unique_clicks"])
+        file_out.write(
+            "\nPercentage Clicks: %f" % campaign_data[level]["percent_clicks"]
+        )
+
+    file_out.close()
+    logging.info("Writing out summary JSON to %s" % campaign_summary_json)
```

Comment:
Use preferred formatting syntax in logging functions:
```suggestion
    logging.info("Writing out summary JSON to %s", campaign_summary_json)
```

Reply:
See 5816a24f37e9a600f2b55946de18b7acb8d0da07